src/solve.jl

```diff
@@ -43,7 +43,12 @@ end
 function DiffEqBase.__solve(prob::DiffEqBase.AbstractSteadyStateProblem,
                             alg::DynamicSS,args...;kwargs...)
 
-  _prob = ODEProblem(prob.f,prob.u0,(0.0,Inf),prob.p)
-  solve(_prob,alg.alg,args...;kwargs...,
-        callback=TerminateSteadyState(alg.abstol,alg.reltol))
+  tspan = alg.tspan isa Tuple ? alg.tspan : (0.0, alg.tspan)
+  _prob = ODEProblem(prob.f,prob.u0,tspan,prob.p)
+  sol = solve(_prob,alg.alg,args...;kwargs...,
+              callback=TerminateSteadyState(alg.abstol,alg.reltol))
+  if sol.t[end] ≈ _prob.tspan[end]
```

Comment:
I feel dumb to do this.  Is `=` safer?  (e.g., when `u0` happened to be a steady state)  Or we need a callback that sets a flag?  Maybe it's better to have a `:Terminated` return code which is set by `terminate!`.

Reply:
`==` is safer. The handling should make this exact.